sas9.4-viya/azuredeploy.json

```diff
@@ -0,0 +1,3683 @@
+                        "name": "[concat(variables('meta_vm_name'),'-osdisk')]",
+                        "createOption": "FromImage",
+                        "caching": "ReadWrite",
+                        "managedDisk": {
+                            "storageAccountType": "Premium_LRS"
+                        },
+                        "diskSizeGB": "[variables('sas_osdisk_size')]"
+                    },
+                    "dataDisks": [
+                        {
+                            "name": "[concat(variables('meta_vm_name'),'-opt-sas')]",
+                            "diskSizeGB": "[variables('sas_opt_sas_disk_size')]",
+                            "lun": 0,
+                            "createOption": "Empty"
+                        }
+                    ]
+                },
+                "osProfile": {
+                    "computerName": "[variables('meta_vm_name')]",
+                    "adminUsername": "[variables('primary_user_name')]",
+                    "linuxConfiguration": {
+                        "disablePasswordAuthentication": true,
+                        "ssh": {
+                            "publicKeys": [
+                                {
+                                    "path": "[concat('/home/', variables('primary_user_name'), '/.ssh/authorized_keys')]",
+                                    "keyData": "[parameters('SSHPublicKey')]"
+                                }
+                            ]
+                        }
+                    }
+                },
+                "networkProfile": {
+                    "networkInterfaces": [
+                        {
+                            "id": "[resourceId('Microsoft.Network/networkInterfaces', variables('meta_nw_interface'))]"
+                        }
+                    ]
+                },
+                "diagnosticsProfile": {
+                    "bootDiagnostics": {
+                        "enabled": true,
+                        "storageUri": "[reference(variables('diagnostic_storagegroup_name')).primaryEndpoints.blob]"
+                    }
+                },
+                "proximityPlacementGroup": {
+                    "id": "[resourceId('Microsoft.Compute/proximityPlacementGroups',variables('ppg_name'))]"
+                }
+            },
+            "resources": []
+        },
+        {
+            "type": "Microsoft.Resources/deployments",
+            "apiVersion": "2019-10-01",
+            "name": "MetaRoleAssignment",
+            "dependsOn": [
+                "[resourceId('Microsoft.Compute/virtualMachines', variables('meta_vm_name'))]"
+            ],
+            "properties": {
+                "mode": "Incremental",
+                "templateLink": {
+                    "uri": "[variables('role_assignment_template')]"
+                },
+                "parameters": {
+                    "roleAssignmentName": {
+                        "value": "[guid('meta',resourceGroup().id)]"

```

Comment:
as above, this needs to include the principal, scope and role

Reply:
done